packages/remix/src/generators/library/library.impl.ts

```diff
@@ -1,55 +1,45 @@
+import type { Tree } from '@nx/devkit';
 import {
   detectPackageManager,
+  formatFiles,
   GeneratorCallback,
-  getWorkspaceLayout,
-  joinPathFragments,
-  names,
-  readProjectConfiguration,
   runTasksInSerial,
-  Tree,
-  updateJson,
-  updateProjectConfiguration,
 } from '@nx/devkit';
 import { Linter } from '@nx/linter';
 import { libraryGenerator } from '@nx/react/src/generators/library/library';
 import { execSync } from 'child_process';
-import { NxRemixGeneratorSchema } from './schema';
+import {
+  addTsconfigEntryPoints,
+  normalizeOptions,
+  updateBuildableConfig,
+} from './lib';
+import type { NxRemixGeneratorSchema } from './schema';
 
-export default async function (tree: Tree, options: NxRemixGeneratorSchema) {
+export default async function (tree: Tree, schema: NxRemixGeneratorSchema) {
   const tasks: GeneratorCallback[] = [];
-  const name = names(options.name).fileName;
   const pm = detectPackageManager();
-  const { libsDir } = getWorkspaceLayout(tree);
-  const projectRoot = joinPathFragments(libsDir, name);
+  const options = normalizeOptions(tree, schema);
 
   const libGenTask = await libraryGenerator(tree, {
-    name,
+    name: options.name,
     style: options.style,
     unitTestRunner: 'jest',
     tags: options.tags,
     importPath: options.importPath,
-    skipFormat: false,
+    directory: options.directory,
+    skipFormat: true,
     skipTsConfig: false,
     linter: Linter.EsLint,
     component: true,
+    buildable: options.buildable,
   });
   tasks.push(libGenTask);
 
-  // Nest dist under project root to we can link it
-  const project = readProjectConfiguration(tree, name);
-  project.targets.build.options = {
-    ...project.targets.build.options,
-    format: ['cjs'],
-    outputPath: joinPathFragments(projectRoot, 'dist'),
-  };
-  updateProjectConfiguration(tree, name, project);
+  addTsconfigEntryPoints(tree, options);
 
-  // Point to nested dist for yarn/npm/pnpm workspaces
-  updateJson(tree, joinPathFragments(projectRoot, 'package.json'), (json) => {
-    json.main = './dist/index.cjs.js';
-    json.typings = './dist/index.d.ts';
-    return json;
-  });
+  if (options.buildable) {
+    updateBuildableConfig(tree, options.projectName);
+  }
 
   // Link workspaces
   tasks.push(() => {
```

Comment:
We have this section that pushes a command to the generator callback tasks to do a Workspace-based install.

This fails when running the generator in a non-(npm/pnpm/yarn) workspace. Does Remix _need_ to be used in a Workspaces-based repo?

Reply:
cc @jaysoo